token/delegation/delegationtest/generator/generator.go

```diff
@@ -203,36 +202,46 @@ func (g *generator) createProofChain(name string, prf []cid.Cid) {
 }
 
 func (g *generator) writeGoFile() error {
-	file := jen.NewFile("delegationtest")
-	file.HeaderComment("Code generated by delegationtest - DO NOT EDIT.")
+	var err error
 
-	refs := map[cid.Cid]string{}
+	f, err := os.Create("../token_gen.go")
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		err = f.Close()
+	}()
+
+	fmt.Fprintln(f, "// Code generated by delegationtest - DO NOT EDIT.")
+	fmt.Fprintln(f)
+	fmt.Fprintln(f, "package delegationtest")
+	fmt.Fprintln(f)
+	fmt.Fprintln(f, "import \"github.com/ipfs/go-cid\"")
+
+	refs := make(map[cid.Cid]string, len(g.dlgs))
 
 	for _, d := range g.dlgs {
 		refs[d.id] = d.name + "CID"
 
-		file.Var().Defs(
-			jen.Id(d.name+"CID").Op("=").Qual("github.com/ipfs/go-cid", "MustParse").Call(jen.Lit(d.id.String())),
-			jen.Id(d.name).Op("=").Id("mustGetDelegation").Call(jen.Id(d.name+"CID")),
-		)
-		file.Line()
+		fmt.Fprintln(f)
+		fmt.Fprintln(f, "var (")
+		fmt.Fprintf(f, "\t%s = cid.MustParse(\"%s\")\n", d.name+"CID", d.id.String())
+		fmt.Fprintf(f, "\t%s    = mustGetDelegation(%s)\n", d.name, d.name+"CID")
+		fmt.Fprintln(f, ")")
 	}
 
 	for _, c := range g.chains {
-		g := jen.CustomFunc(jen.Options{
-			Multi:     true,
-			Separator: ",",
-			Close:     "\n",
-		}, func(g *jen.Group) {
-			slices.Reverse(c.prf)
-			for _, p := range c.prf {
-				g.Id(refs[p])
-			}
-		})
+		fmt.Fprintln(f)
+		fmt.Fprintf(f, "var %s = []cid.Cid{\n", c.name)
+
+		slices.Reverse(c.prf)
```

Comment:
you could use `slices.Backward()` to iterate in reverse without actually changing the data